### DISCUSSION TITLE
Better Altitude difference calculation

### DISCUSSION THREAD

**Discussion by AstAst42** [organicmaps/organicmaps#1226] [Ideas]
The calculation for the altitude difference seems a bit weird, mostly because most other hiking/biking apps calculate it differently. At the moment the calculation gives you the difference between the highest and lowest point in your route but I think it should combine all "uphill / downhill distances" like other apps. For example if I hike/bike 2 mountains both 500m high, starting at altitude 0, it should say at least 1000m and not 500m like it is now.

**Comment by biodranik** [comment]
The shape of the path should show you how many hills you have, no? Also, some users may think that 1000m in your example is the height of every mountain. Any ideas on how to avoid this confusion?

Also, it would be great to see some screenshots from other apps with intuitive graphs.

**Reply by AstAst42** [reply]
The graph itself is fine but the calculation for the altitude number right above the graph is weird. I can't imagine that some users would think that each mountain would be 1000m because the graph is right beside the number and the graph got its own numbers where you can see this (and you can look up mountain heights by tapping on them in the normal map). So the calculated number feels just weird and doesn't fit to the graph. 
I upload some images of other apps and all of them calculate the summarized ascent. (some of them are tracker apps, so not for planning, some of them are made for round trips only so up- and downhill is the same number and shown as one, some are for planning just like organic maps and simply show up- & downhill separately, I also put one image of organic maps with a marker for a good spot for the seperat values and sorry for the German Screenshots, but I marked the important values) 

![pic_01](https://user-images.githubusercontent.com/34718776/131748302-eb49d49f-dbdd-42a2-8306-0c9d23553493.jpg)
![pic_02](https://user-images.githubusercontent.com/34718776/131748305-2636dd88-b71c-4ffa-a0db-9b2c0502ca0d.jpg)
![pic_03](https://user-images.githubusercontent.com/34718776/131748308-4d2aa489-97c8-4201-912e-aa0479166e7b.jpg)
![pic_00](https://user-images.githubusercontent.com/34718776/131748311-fb04f14a-e141-4d65-ab0e-2c550a6ebf0e.jpg)
![pic_04](https://user-images.githubusercontent.com/34718776/131748313-cfcae7e3-a993-4432-a20c-4126dad90dc5.jpg)
![pic_06](https://user-images.githubusercontent.com/34718776/131748315-f4148026-9e66-4839-91bb-58cc7d4d7d7e.jpg)
![pic_05](https://user-images.githubusercontent.com/34718776/131748317-06cad6a7-7f68-4087-a258-4b2b7b517003.jpg)

**Reply by 0xD0M1M0** [reply]
I saw the feature integrated in one of the latest versions. I kind of like it, but it is not helpful in some cases.

In my case, a rather flat 1.4km way has a lot of ups and downs making it a bit wired.

![Screenshot_20221104-205349.png](https://user-images.githubusercontent.com/76812428/200064365-57201389-97ec-4a06-ab38-92ed9beb4f43.png)

Especially when hiking in the mountains, it is useful to know the lowest and the highest point of the tour as well and not just the difference. That would give a clear feedback in which altitude the map is located and how the chart should be understood.
